### PR TITLE
[AF-1775] SSH Ciphers and MAC configured via args on startup

### PR DESCRIPTION
**Pull request by desmax74**
@porcelli  @Rikkola @ederign 

docs:https://github.com/kiegroup/kie-docs/pull/1342
https://github.com/kiegroup/kie-docs/pull/1346

### CODE REVIEW

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 1]
```diff
@@ -51,26 +55,64 @@
...
+                        setUpBuiltinFactories(false,
+                                              managedMACs)).build();
+    }
 
-        return builder().cipherFactories(setUpBuiltinFactories(false,
-                                                               ciphers)).build();
+    private SshServer buildSshServer(List<BuiltinCiphers> ciphersConfigured,
+                                     List<BuiltinMacs> macsConfigured) {
+        ServerBuilder serverBuilder = builder();
+        if(!ciphersConfigured.isEmpty()){
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: this code should be simplified, the parameter should be a predicate to filter the list, so we don't have to mantain 2 lists and check if input is empty.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 2]
```diff
@@ -91,14 +135,20 @@ public void setup(final File certDir,
...
+        sshd.getProperties().put(SshServer.IDLE_TIMEOUT,
+                                 sshIdleTimeout);
 
         if (inetSocketAddress != null) {
             sshd.setHost(inetSocketAddress.getHostName());
             sshd.setPort(validateOrGetNew(inetSocketAddress.getPort()));
 
             if (inetSocketAddress.getPort() != sshd.getPort()) {
-                LOG.error("SSH for Git original port {} not available, new free port {} assigned.", inetSocketAddress.getPort(), sshd.getPort());
+                LOG.error("SSH for Git original port {} not available, new free port {} assigned.",
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: avoid change format where you don't need

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 3]
```diff
@@ -137,29 +187,101 @@ public void setup(final File certDir,
         });
         sshd.setPublickeyAuthenticator(new CachingPublicKeyAuthenticator((username, key, session) -> {
 
-            final FileSystemUser user = getSshAuthenticator().authenticate(username, key);
+            final FileSystemUser user = getSshAuthenticator().authenticate(username,
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: avoid change format where you don't need

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 4]
```diff
@@ -137,29 +187,101 @@ public void setup(final File certDir,
...
+                                                                           key);
 
             if (user == null) {
                 return false;
             }
 
-            session.setAttribute(BaseGitCommand.SUBJECT_KEY, user);
+            session.setAttribute(BaseGitCommand.SUBJECT_KEY,
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: avoid change format where you don't need

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 5]
```diff
@@ -137,29 +187,101 @@ public void setup(final File certDir,
...
+                                 user);
 
             return true;
         }));
         sshd.setPasswordAuthenticator((username, password, session) -> {
 
-            final FileSystemUser user = getUserPassAuthenticator().authenticate(username, password);
+            final FileSystemUser user = getUserPassAuthenticator().authenticate(username,
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: avoid change format where you don't need

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 6]
```diff
@@ -137,29 +187,101 @@ public void setup(final File certDir,
...
+                                                                                password);
 
             if (user == null) {
                 return false;
             }
 
-            session.setAttribute(BaseGitCommand.SUBJECT_KEY, user);
+            session.setAttribute(BaseGitCommand.SUBJECT_KEY,
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: avoid change format where you don't need

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 7]
```diff
@@ -137,29 +187,101 @@ public void setup(final File certDir,
...
             return true;
         });
     }
 
+    private void buildSSHServer(String gitSshCiphers,
+                                String gitSshMacs) {
+
+        List<BuiltinCiphers> ciphers = checkAndSetGitCiphers(gitSshCiphers);
+        List<BuiltinMacs> macs = checkAndSetGitMacs(gitSshMacs);
+        if (ciphers.isEmpty() || macs.isEmpty()) {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: commented about predicate/filter list above

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 8]
```diff
@@ -184,6 +306,14 @@ SshServer getSshServer() {
         return sshd;
     }
 
+    public List<String> getCipherFactoriesNames(){
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: don't see a need for this

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 9]
```diff
@@ -184,6 +306,14 @@ SshServer getSshServer() {
...
+        return sshd.getCipherFactoriesNames();
+    }
+
+    public List<String> getMacFactoriesNames(){
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: don't see a need for this

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 8 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Is for read the configured ciphers, used in the test

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 9 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Is for read the configured Macs, used in the test

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 6 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 5 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 2 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 3 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 4 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java [hunk 10]
```diff
@@ -439,4 +449,8 @@ public TimeUnit getDefaultJgitCacheEvictThresholdTimeUnit() {
     public long getJgitCacheEvictThresholdDuration() {
         return jgitCacheEvictThresholdDuration;
     }
+
+    public String getGitSshCiphers() { return gitSshCiphers; }
+
+    public String getGitSshMACS() { return gitSshMACS; }
```
[the remark this answers is not in the document]
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java [hunk 11]
```diff
@@ -112,6 +114,8 @@
     private String sshAlgorithm;
     private String sshPassphrase;
     private String sshIdleTimeout;
+    private String gitSshCiphers;
+    private String gitSshMACS;
```
[the remark this answers is not in the document]
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 1 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 12]
```diff
@@ -51,26 +59,41 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(GitSSHService.class);
 
-    private final SshServer sshd = buildSshServer();
-
+    private static List<BuiltinCiphers> managedCiphers =
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: not big fan of static here, but at least add final to it

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 13]
```diff
@@ -51,26 +59,41 @@
...
+                    BuiltinCiphers.aes128ctr,
+                    BuiltinCiphers.aes192ctr,
+                    BuiltinCiphers.aes256ctr,
+                    BuiltinCiphers.arcfour256,
+                    BuiltinCiphers.arcfour128,
+                    BuiltinCiphers.aes192cbc,
+                    BuiltinCiphers.aes256cbc
+            ));
+
+    private static List<BuiltinMacs> managedMACs =
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: not big fan of static here, but at least add final to it

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java [hunk 14]
```diff
@@ -234,7 +252,9 @@ public void testCheckSetupParameters() throws Exception {
                              "RSA",
                              null,
                              mock(JGitFileSystemProvider.RepositoryResolverImpl.class),
-                             executorService);
+                             executorService,
+                             null,
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: not liking all these nulls here. if preserve the original tests without params, I'd prefer to add an overload constructor

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java [hunk 15]
```diff
@@ -260,9 +282,287 @@ public void testCheckSetupParameters() throws Exception {
...
+                         null);
+
+        sshService.start();
+        assertTrue(sshService.isRunning());
+
+        List<String> ciphersReaded = sshService.getSshServer().getCipherFactoriesNames();
+        List<String> macsReaded = sshService.getSshServer().getMacFactoriesNames();
+
+        assertThat(ciphersReaded).hasSize(7);
+        assertThat(ciphersReaded.containsAll(Arrays.asList("aes128-ctr",
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: this test is fragile, better before iterate over all available ciphers and macs to make sure that in a upgrade of library we don't break the test

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java (hunk 14 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 13 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 12 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java (hunk 15 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java (hunk 14 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 13 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 7 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 16]
```diff
@@ -51,26 +53,43 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(GitSSHService.class);
 
-    private final SshServer sshd = buildSshServer();
-
+    //aes128-ctr, aes192-ctr, aes256-ctr, arcfour128, arcfour256, aes192-cbc, aes256-cbc
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: this comment looks unnecessary

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 17]
```diff
@@ -51,26 +53,43 @@
...
+                    BuiltinCiphers.aes128ctr,
+                    BuiltinCiphers.aes192ctr,
+                    BuiltinCiphers.aes256ctr,
+                    BuiltinCiphers.arcfour256,
+                    BuiltinCiphers.arcfour128,
+                    BuiltinCiphers.aes192cbc,
+                    BuiltinCiphers.aes256cbc
+            ));
+
+    // hmac-md5, hmac-md5-96, hmac-sha1, hmac-sha1-96, hmac-sha2-256, hmac-sha2-512
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: this comment looks unnecessary

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 18]
```diff
@@ -160,6 +195,53 @@ public void setup(final File certDir,
...
+        } else {
+            List<BuiltinCiphers> ciphersHandled = new ArrayList<>();
+            List<String> ciphers = Arrays.asList(gitSshCiphers.split(","));
+            for (String cipherCode : ciphers) {
+                BuiltinCiphers cipher = BuiltinCiphers.fromFactoryName(cipherCode.trim().toLowerCase());
+                if (cipher != null && managedCiphers.contains(cipher)) {
+                    ciphersHandled.add(cipher);
+                    LOG.info("Added Cipher {} to the git ssh configuration. ", cipher);
+                }else{
+                    LOG.info("Cipher {} not handled in git ssh configuration. ", cipher);
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: maybe raise this to WARN

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 19]
```diff
@@ -160,6 +195,53 @@ public void setup(final File certDir,
...
+
+            List<BuiltinMacs> macs = new ArrayList<>();
+            List<String> macsInput = Arrays.asList(gitSshMacs.split(","));
+            for (String macCode : macsInput) {
+                BuiltinMacs mac = BuiltinMacs.fromFactoryName(macCode.trim().toLowerCase());
+                if (mac != null && managedMACs.contains(mac)) {
+                    macs.add(mac);
+                    LOG.info("Added MAC {} to the git ssh configuration. ", mac);
+                }else{
+                    LOG.info("MAC {} not handled in git ssh configuration. ", mac);
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: maybe raise this to WARN

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 18 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 19 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Ok

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 17 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: We want add this in the docs ?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 16 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: We want add this in the docs ?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 17 above)
Comment (porcelli) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Docs will certainly need to be updated

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 16 above)
Comment (porcelli) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Docs will certainly need to be updated

**Comment by desmax74** [comment]
@porcelli fixed

**Comment by porcelli** [comment]
let's wait for build and qe to test to get this merged.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java [hunk 20]
```diff
@@ -75,6 +75,8 @@
     public static final String GIT_ENV_KEY_INIT = "init";
     public static final String GIT_ENV_KEY_MIRROR = "mirror";
     public static final String SCHEME = "git";
+    public static final String GIT_SSH_CIPHERS = "org.uberfire.nio.git.ssh.ciphers";
```
Comment (jhrcek) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java]: Please don't forget to update the property name in kie-docs

**Comment by jhrcek** [comment]
@desmax74 tested locally and works fine. Please update kie-docs with the new property name and fix the test. After that I'll approve this PR.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java [hunk 21]
```diff
@@ -265,4 +270,234 @@ public void testCheckSetupParameters() throws Exception {
...
+        assertThat(sshService.getSshServer().getProperties().get(SshServer.IDLE_TIMEOUT)).isEqualTo(idleTimeout);
+
+        sshService.stop();
+
+        assertFalse(sshService.isRunning());
+    }
+
+    private void checkCiphersName(List<String> ciphersReaded){
+        for(String cipher : ciphersReaded){
+            assertThat(BuiltinCiphers.fromFactoryName(cipher) != null);
```
[the remark this answers is not in the document]
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: Thanks for the refresh on AssertJ :)  fixed

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java (hunk 20 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConfiguration.java]: Yep, https://github.com/kiegroup/kie-docs/pull/1346 :)

**Pull request merged**